tools/src/gt_mpi_gather.cc

```diff
@@ -385,19 +384,37 @@ void produce_column_histogram(const VariantQueryProcessor& qp, const VariantQuer
     histogram_op.equi_partition_and_print_bins(val);
 }
 
-int main(int argc, char *argv[]) {
-  //Initialize MPI environment
+void print_usage() {
+  std::cout << "Usage: gt_mpi_gather [options]\n"
+            << "where options include:\n"
+            << "\t \e[1m--help\e[0m, \e[1m-h\e[0m Print a usage message summarizing options available and exit\n"
+            << "\t \e[1m--json-config\e[0m=<query json file>, \e[1m-j\e[0m <query json file>\n"
+            << "\t\t Can specify workspace, array, query_column_ranges, query_row_ranges, vid_mapping_file,\n"
+            << "\t\t callset_mapping_file, query_attributes, query_filter, reference_genome, etc. as fields in the json file\n"
+            << "\t \e[1m--loader-json-config\e[0m=<loader json file>, \e[1m-l\e[0m <loader json file>\n"
+            << "\t\t Optional, if vid_mapping_file and callset_mapping_file fields are specified in the query json file\n"
```

Comment:
Not an expert on overall usage here (I only use a few of these myself...) but one comment: should we make the required/optional a bit clearer somehow? I only ask since we explicitly mention loader-json-config as optional if vid/callset are specified in query.

Reply:
Agreed. I am not an expert on the usage either. These are just to get started. I am relying on @mlathara and @kgururaj to get these completed.

Reply:
How does this look?
```
Usage: gt_mpi_gather [options]
where options include:
	 --help, -h Print a usage message summarizing options available and exit
	 --json-config=<query json file>, -j <query json file>
		 Can specify workspace, array, query_column_ranges, query_row_ranges, vid_mapping_file,
		 callset_mapping_file, query_attributes, query_filter, reference_genome, etc. as fields in the json file e.g.
			 { "workspace" : "/tmp/ws",
			   "array" : "t0_1_2",
			   "query_column_ranges" : [ [ [0, 100 ], 500 ] ],
			   "query_row_ranges" : [ [ [0, 2 ] ],
			   "vid_mapping_file" : "/tests/inputs/vid.json",
			   "query_attributes" : [ "REF", "ALT", "BaseQRankSum", "MQ", "MQ0", "ClippingRankSum", "MQRankSum", "ReadPosRankSum", "DP", "GT", "GQ", "SB", "AD", "PL", "DP_FORMAT", "MIN_DP" ] }
	 --loader-json-config=<loader json file>, -l <loader json file>
		 Optional, if vid_mapping_file and callset_mapping_file fields are specified in the query json file
	 --workspace=<workspace dir>, -w <GenomicsDB workspace dir>
		 Optional, if workspace is specified in any of the json config files
	 --array=<array dir>, -A <GenomicsDB array dir>
		 Optional, if array is specified in any of the json config files
	 --print-calls
		 Optional, prints VariantCalls in a JSON format
	 --print-csv
		 Optional, outputs CSV with the fields and the order of CSV lines determined by the query attributes
	 --produce-Broad-GCVF
		 Optional, produces combined gVCF from the GenomicsDB data constrained by the query configuration
		 --output-format=<output_format>, -O <output_format>
			 used with --produce-Broad-GCVF
			 Output format can be one of the following strings: "z[0-9]" (compressed VCF),"b[0-9]" (compressed BCF)
			 or "bu" (uncompressed BCF). Default is uncompressed VCF if not specified.
	 --produce-histogram
		 Optional
	 --produce-interesting-positions
		 Optional
	 --version Print version and exit
	If none of the print/produce arguments are specified, the tool prints all the Variants constrained by the query configuration in a JSON format

	Parallel Querying
		 MPI could be used for parallel querying, e.g.
		 mpirun -n <num_processes> -hostfile <hostfile> ./bin/gt_mpi_gather -j <query.json> -l <loader.json> [<other_args>]
```

Reply:
I think this looks pretty good! 

One note: the usage message still says `--produce-Broad-**GCVF**` instead of `produce-Broad-**GVCF**`

Reply:
Ah, the dyslexic me :-). Will correct.